src/components/Post/CommentDialogBox.jsx

```diff
@@ -68,14 +69,15 @@ const CommentDialogBox = ({
                     <ReadMore>{userComment.content.text}</ReadMore>
                   </p>
                 </div>
+
                 <div
                   onClick={() => {
                     setOpenToDeleteComment(!openToDeleteComment);
                     setDeleteCommentID(userComment);
                   }}
                 >
                   {user &&
-                    userComment.content.username === user.displayName && (
+                    userComment?.content?.displayName == user?.displayName && (
```

Comment:
again displaynames are not unique and having someone else delete your comment won't look nice @biplavmz . We can change our display name to anyone in edit profile option but not username

Reply:
try to only compare by username

Reply:
@narayan954 bro previously i compare with the username only because it's is unique and user.uid is unique for every user 
userComment?.content?.username === user.uid

Reply:
but it's not the same, uid!==username

Reply:
ya @narayan954  bro it is same userComment?.content?.username === user.uid that's why it's working fine.

Reply:
nope not always

Reply:
@narayan954  can we compare like this 
userComment.content.username === auth?.currentUser?.uid
like how it was implemented in Imageupload component

Reply:
nope, I am again telling you, username is not always UID, username can be edited, UID is the document uid given by firestore database. 
When you're creating an account using oauth, you're given a default username as your uid, but that can be changed anytime in the profile edit option. When creating account using email method, you choose your username by yourself! and hence there's no chance we can assume username is equal to uid. 
Since there's a user collection now in db, I'd suggest you to follow the approach as in imgupload component to get the username for the current user and then compare it with the one on comments.

I told this all before as a hint in the first comment only of our previous conversation. And there's nowhere `userComment.content.username === auth?.currentUser?.uid` implemented in Imageupload component and if it was it must have been a mistake.